extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/StaticInitHibernatePersistenceProviderResolver.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.hibernate.orm.runtime;
+
+import java.util.List;
+
+import javax.persistence.spi.PersistenceProvider;
+import javax.persistence.spi.PersistenceProviderResolver;
+
+/**
+ * During the static init phase, we don't access the PersistenceProviderResolver.
+ */
+final class StaticInitHibernatePersistenceProviderResolver implements PersistenceProviderResolver {
```

Comment:
I'd love to understand the need for this better. Could you elaborate, and maybe evolve the comments for future reference?

Reply:
This is something we shouldn't access during the static init phase as it's only used when creating the entity manager factory.

Thus, to be extra sure we don't access it, I decided to make it extra safe by throwing exceptions.

We now define the `PersistenceProviderResolver` twice: once at the beginning of the static init phase with this one, which shouldn't be used and the second time at the beginning of the runtime phase where we also push the runtime properties.

BTW, it was more or less done this way before i.e. I haven't changed anything really, just made it clearer. Mostly because I now need to push the runtime properties in the constructor. I could have define a setter and rely on the order of operations but I prefer having things immutable and fully constructed.